gramineproject/gramine
# [python] Add support for production signing with Azure Key Vault

Signed-off-by: Sankaranarayanan Venkatasubramanian <sankaranarayanan.venkatasubramanian@intel.com>



## Description of the changes 
This change enables production signing with Azure Key Vault's Managed HSM.
https://github.com/gramineproject/gsc/pull/108 has corresponding gsc changes.


Fixes #11 (for AKV)
Fixes #698 (for AKV)


## How to test this PR? 

- You need to have Azure Subscription with access to Azure Key Vault's Managed HSM.
- Create a 3072-bit RSA key in Azure Key Vault's managed HSM with public exponent of 3 (Please note that only Managed HSM supports setting public exponent) and give permissions to yourself or who will use the key created as 'Managed HSM Crypto User' (az keyvault role assignment).
- `gramine-sgx-sign` is modified to take a keytype parameter that will use `pem` for local private key and `akv` for Azure Key Vault. Please see updated documentation.

This was tested with my Azure Key Vault subscription, and @dimakuv reviewed it as part of our pair review (Thanks, @dimakuv)


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/1020)
